flink-runtime/src/main/java/org/apache/flink/runtime/rest/handler/legacy/metrics/AbstractAggregatingMetricsHandler.java

```diff
@@ -0,0 +1,223 @@
+					} else {
+						for (String aggregation : aggTypeList.split(",")) {
+							switch (aggregation.toLowerCase()) {
+								case DoubleAccumulator.DoubleMinimum.NAME:
+									requestedAggregationsFactories.add(DoubleAccumulator.DoubleMinimumFactory.get());
+									break;
+								case DoubleAccumulator.DoubleMaximum.NAME:
+									requestedAggregationsFactories.add(DoubleAccumulator.DoubleMaximumFactory.get());
+									break;
+								case DoubleAccumulator.DoubleSum.NAME:
+									requestedAggregationsFactories.add(DoubleAccumulator.DoubleSumFactory.get());
+									break;
+								case DoubleAccumulator.DoubleAverage.NAME:
+									requestedAggregationsFactories.add(DoubleAccumulator.DoubleAverageFactory.get());
+									break;
+								default:
+									log.warn("Invalid aggregation specified: {}", aggregation.toLowerCase());
+							}
+						}
+					}
+					Collection<? extends MetricStore.ComponentMetricStore> stores = getStores(store, pathParams, queryParams);
+					if (stores == null){
+						return "[]";
+					}
+					return getAggregatedMetricValues(stores, requestedMetrics, requestedAggregationsFactories);
+				} catch (Exception e) {
+					throw new CompletionException(new FlinkException("Could not retrieve metrics.", e));
+				}
+			},
+			executor);
+	}
+
+	/**
+	 * Returns a JSON string containing a list of all available metrics in the given stores. Effectively this method maps
+	 * the union of all key-sets to JSON.
+	 *
+	 * @param stores metrics
+	 * @return JSON string containing a list of all available metrics
+	 */
+	private static Collection<String> getAvailableMetrics(Collection<? extends MetricStore.ComponentMetricStore> stores) {
+		Set<String> uniqueMetrics = new HashSet<>();
+		for (MetricStore.ComponentMetricStore store : stores) {
+			uniqueMetrics.addAll(store.metrics.keySet());
+		}
+		return uniqueMetrics;
+	}
+
+	private static String mapMetricListToJson(Collection<String> metrics) throws IOException {
+		StringWriter writer = new StringWriter();
+		JsonGenerator gen = JsonFactory.JACKSON_FACTORY.createGenerator(writer);
+
+		gen.writeStartArray();
+		for (String m : metrics) {
+			gen.writeStartObject();
+			gen.writeStringField("id", m);
+			gen.writeEndObject();
+		}
+		gen.writeEndArray();
+
+		gen.close();
+		return writer.toString();
+	}
+
+	/**
+	 * Extracts and aggregates all requested metrics from the given metric stores, and maps the result to a JSON string.
+	 *
+	 * @param stores available metrics
+	 * @param requestedMetrics ids of requested metrics
+	 * @param requestedAggregationsFactories requested aggregations
+	 * @return JSON string containing the requested metrics
+	 * @throws IOException
+	 */
+	private String getAggregatedMetricValues(Collection<? extends MetricStore.ComponentMetricStore> stores, String[] requestedMetrics, List<DoubleAccumulator.DoubleAccumulatorFactory<?>> requestedAggregationsFactories) throws IOException {
```

Comment:
This is a personal style thing but I would break the line of the arguments into multiple lines, with one argument each.